src/main/java/it/gov/pagopa/payment/instrument/service/idpaycode/EncryptCodeServiceImpl.java

```diff
@@ -1,18 +1,81 @@
 package it.gov.pagopa.payment.instrument.service.idpaycode;
 
+import it.gov.pagopa.payment.instrument.exception.PaymentInstrumentException;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.util.Base64;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.codec.binary.Hex;
+import org.apache.commons.lang3.StringUtils;
 import org.springframework.stereotype.Service;
 
 @Service
 @Slf4j
 public class EncryptCodeServiceImpl implements EncryptCodeService {
 
-  public EncryptCodeServiceImpl() {
-    // TODO to be implemented
-  }
+  public static final String GENERATE_PIN_BLOCK = "GENERATE_PIN_BLOCK";
+  public static final String HASH_PIN_BLOCK = "HASH_PIN_BLOCK";
 
   @Override
-  public String encryptIdpayCode(String code) {
-    return code;
+  public String encryptIdpayCode(String code, String secondFactor, String salt) {
```

Comment:
let's use more meaningful names
```suggestion
  public String buildHashedPinBlock(String code, String secondFactor, String salt) {
```